crates/telio-proxy/src/proxy.rs

```diff
@@ -191,6 +243,40 @@ impl StateEgress {
 
         Ok(())
     }
+
+    fn log_error(err: &Error) {
+        match err {
+            Error::IoError(e) => telio_log_error!("Unable to send. Error {}", e.kind()),
+            _ => telio_log_error!("Unable to send. Error {}", err),
+        }
+    }
+
+    async fn handle_error(&mut self, err: Error) {
+        if self.conn_state != OutboundConnState::Error {
+            Self::log_error(&err);
+
+            match ErrorType::from(err) {
+                ErrorType::RecoverableError => (),
+                ErrorType::UnrecoverableError => {
+                    Self::sleep_forever().await;
+                }
+                ErrorType::SocketIOError => {
+                    if let Err(err) = SocketPool::new_udp(
+                        SocketAddr::from((Ipv4Addr::LOCALHOST, 0)),
+                        Some(UdpParams(SocketBufSizes {
+                            rx_buf_size: Some(SOCK_BUF_SZ),
+                            tx_buf_size: Some(SOCK_BUF_SZ),
+                        })),
+                    )
+                    .await
+                    {
+                        telio_log_error!("Cannot open socket. ERR :{}", err);
+                    };
+                }
+            }
+            self.conn_state = OutboundConnState::Error;
```

Comment:
If we have a Recoverable/SocketIO-Unrecoverable error sequence it will still call recv(). I don't think `conn_state` is needed here

Reply:
Yes. recv() would be called again and again and until the error is resolved it will keep logging error. conn_state is added to avoid spamming the logs with error. That applies for when everything is working as well!

Reply:
If the channel is broken, it doesn't matter how many times you call `recv()`, it won't fix it.

Let's say you have a Recoverable/SocketIO error (`conn_state` -> `OutboundConnState::Error`) and then, right on the next call to `recv()` the channel is broken and an Unrecoverable error is thrown. How are you going to handle the error, because if `conn_state` is already on the `OutboundConnState::Error` state it won't enter the if clause from [line 239](https://github.com/NordSecurity/libtelio/pull/221/files#diff-9c4504f1cf7958587fd86df0679204014595a9d76b01c8420e61c442cb4a1dd7R239).

Reply:
Mmhm. Makes sense!
I changed the logic a bit. Made conn_state as result.